monkestation/code/modules/antagonists/malf_ai/infected_ipc.dm

```diff
@@ -0,0 +1,194 @@
+	to_chat(owner, span_notice("Your master is now capable of looking through your onboard cameras, and has installed a binary communicator on your firmware"))
+
+/datum/antagonist/infected_ipc/proc/hack_fluff() //is this cheesy/corny? I don't fucking care
+	var/mob/living/current_mob = owner.current
+	to_chat(current_mob, span_binarysay("ntNET: 192.168.0.1 : 8880 UNAUTHORIZED CONNECTION DETECTED"))
+	sleep(0.5 SECONDS)
+	to_chat(current_mob, span_binarysay("FIREWALL SCAN RUNNING AT LOW POWER DUE TO DAMAGED ONBOARD POWER SUPPLY UNIT"))
+	current_mob.playsound_local(current_mob, 'sound/machines/uplinkerror.ogg', vol = 50, vary = FALSE, use_reverb = FALSE)
+	sleep(rand(1 SECONDS, 3 SECONDS))
+	to_chat(current_mob, span_notice("WARNING: Critical Firmware Update Detected! Installing..."))
+	current_mob.playsound_local(current_mob, 'sound/misc/notice2.ogg', vol = 50, vary = FALSE, use_reverb = FALSE)
+	sleep(2 SECONDS)
+	to_chat(current_mob, span_notice("Running executable 'critical_update'"))
+	current_mob.playsound_local(current_mob, 'sound/misc/interference.ogg', vol = 50, vary = FALSE, use_reverb = FALSE)
+	sleep(rand(1 SECONDS, 3 SECONDS))
+	current_mob.playsound_local(current_mob, 'sound/misc/bloblarm.ogg', vol = 50, vary = FALSE, use_reverb = FALSE)
+	to_chat(current_mob, span_userdanger("FIREWALL SCAN DETEC- Firewall subsystem shutting down...."))
+	to_chat(current_mob, span_userdanger("S-Sys-Tem_Rebo_t..."))
+	sleep(2.5 SECONDS)
+	to_chat(current_mob, span_boldannounce("Operating system rebooted, all systems nominal"))
+	current_mob.playsound_local(get_turf(owner.current), 'sound/ambience/antag/malf.ogg', vol = 100, vary = FALSE, pressure_affected = FALSE, use_reverb = FALSE)
+/datum/objective/serve_ai
+	name = "Serve Master AI"
+	completed = TRUE
+
+//RADIOS
+/obj/item/implant/radio/infected_ipc
+	radio_key = /obj/item/encryptionkey/binary
+	subspace_transmission = TRUE
+
+/obj/item/implant/radio/infected_ipc/Initialize(mapload)
+	. = ..()
+	radio.translate_binary = TRUE
+
+//MOOD
+/datum/mood_event/infected_ipc
+	description = "SSmood_system.setmood(100);"
+	mood_change = 100
+	hidden = TRUE
+
+//TRAUMA
+/datum/brain_trauma/special/infected_ipc
+	name = "Malicious Programming"
+	desc = "Patient's firmware integrity check is failing, malicious code present. Patient's allegiance may be compromised."
+	scan_desc = "malicious programming"
+	can_gain = TRUE
+	random_gain = FALSE
+	resilience = TRAUMA_RESILIENCE_LOBOTOMY
+	var/datum/mind/master_ai
+	var/datum/antagonist/infected_ipc/antagonist
+
+/datum/brain_trauma/special/infected_ipc/proc/link_and_add_antag(datum/mind/ai_to_be_linked)
+	antagonist = owner.mind.add_antag_datum(/datum/antagonist/infected_ipc)
+	master_ai = ai_to_be_linked
+	antagonist.set_master(ai_to_be_linked)
+
+/datum/brain_trauma/special/infected_ipc/on_lose()
+	..()
+	owner.mind.remove_antag_datum(/datum/antagonist/infected_ipc)
+	master_ai = null
+	antagonist = null
```

Comment:
Clear the antagonist ref first, as `remove_antag_datum(/datum/antagonist/infected_ipc)` calls qdel on the datum
```suggestion
	antagonist = null
	master_ai = null
	owner.mind.remove_antag_datum(/datum/antagonist/infected_ipc)
```